hacks/installer.nsi

```diff
@@ -0,0 +1,293 @@
+
+Function pgSettingsPageCreate
+    ; migrate old settings
+    IfFileExists $COMMONPROGRAMDATA\myOpenFactory\Client\config.properties 0 skipMigrateConfig
+        DetailPrint "Migrating old configuration."
+        CopyFiles $COMMONPROGRAMDATA\myOpenFactory\Client\* $SettingsDir
+        ${If} ${Errors}
+            DetailPrint "Failed to copy old config!"
+        ${Else}
+          RMDir /r $COMMONPROGRAMDATA\myOpenFactory\Client
+        ${EndIf}
+        nsExec::ExecToLog '"$INSTDIR\edi-client.exe" config migrate $COMMONPROGRAMDATA\myOpenFactory\Client\config.properties $SettingsDir\config.properties'
+    skipMigrateConfig:
+
+    IfFileExists "$SettingsDir\config.properties" 0 +2
+        Abort
+
+    !insertmacro MUI_HEADER_TEXT "EDI-Client Settings" "Provide EDI-Client configuration."
+
+    nsDialogs::Create 1018
+    Pop $Dialog
+
+    ${If} $Dialog == error
+        Abort
+    ${EndIf}
+
+    ${NSD_CreateGroupBox} 5% 5% 90% 60% "Credentials"
+    Pop $0
+
+    ${NSD_CreateLabel} 15% 16% 20% 10u "Username:"
+    Pop $0
+
+    ${NSD_CreateText} 40% 15% 30% 12u ""
+    Pop $UserHWND
+
+    ${NSD_CreateLabel} 15% 31% 20% 10u "Password:"
+    Pop $0
+
+    ${NSD_CreatePassword} 40% 30% 30% 12u ""
+    Pop $PasswordHWND
+
+    ${NSD_CreateLabel} 15% 46% 20% 10u "Certificate:"
+    Pop $0
+
+    ${NSD_CreateFileRequest} 40% 45% 30% 12u \
+        ""
+    Pop $CertificateHWND
+
+    ${NSD_CreateBrowseButton} 75% 45% 10% 12u "Browse"
+    Pop $0
+    ${NSD_OnClick} $0 onCertificateBrowse
+
+    nsDialogs::Show
+FunctionEnd
+
+Function pgSettingsPageLeave
+    ${NSD_GetText} $UserHWND $0
+    StrCpy $Username "$0"
+
+    ${NSD_GetText} $PasswordHWND $0
+    StrCpy $Password "$0"
+
+    ${NSD_GetText} $CertificateHWND $0
+    StrCpy $CertificateFile "$0"
+FunctionEnd
+
+Function onCertificateBrowse
+    ${NSD_GetText} $CertificateHWND $0
+    nsDialogs::SelectFileDialog open "" ".pem|*.pem"
+    Pop $0
+    ${If} $0 != error
+        ${NSD_SetText} $CertificateHWND "$0"
+    ${EndIf}
+FunctionEnd
+
+Function pgServiceSettingsPageCreate
+    ${Unless} ${SectionIsSelected} ${ServiceSection}
+        Abort
+    ${EndUnless}
+
+    !insertmacro MUI_HEADER_TEXT "Service Settings" "Provide service configuration."
+
+    nsDialogs::Create 1018
+    Pop $Dialog
+
+    ${If} $Dialog == error
+        Abort
+    ${EndIf}
+
+    ${NSD_CreateGroupBox} 5% 5% 90% 90% "Service Credentials (Optional)"
+    Pop $0
+
+    ${NSD_CreateLabel} 15% 16% 20% 10u "Username:"
+    Pop $0
+
+    ${NSD_CreateText} 40% 15% 30% 12u ""
+    Pop $UserHWND
+
+    ${NSD_CreateLabel} 15% 26% 20% 10u "Password:"
+    Pop $0
+
+    ${NSD_CreatePassword} 40% 25% 30% 12u ""
+    Pop $PasswordHWND
+
+    ${NSD_CreateLink} 15% 85% 90% 12u "Information about Usernames"
+    Pop $0
+    ${NSD_OnClick} $0 onUsernameInfoClick
+
+    nsDialogs::Show
+FunctionEnd
+
+Function onUsernameInfoClick
+
+ExecShell "open" "https://docs.myopenfactory.com/edi/protocols/edi-connector/#7fehleranalyse"
```

Comment:
After the documentation was written this needs to be updated.

Reply:
@BlackReloaded I think it's fine to change this Link in a followup CL. I would like to work on other things, which depend on this PR.